### DISCUSSION TITLE
How to lookup/list other objects in a webhook callback?

### DISCUSSION THREAD

**Discussion by 13126690189** [kubernetes-sigs/kubebuilder#2433] [Support] [answered]
1. My resource has a field in the spec that references an object of another kind.
While creating an object of my resource type, I need to ensure that the object it references does exist. How to do this?

2. My resource may be referenced by many objects of another kind.
Before deleting an object of my resource type, I need to ensure that it's no longer referenced by any other object. How to do this?

Thanks a lot!

**Comment by camilamacedo86** [comment, marked as answer]
It seems that you are looking for

- Validation During Creation: They want to ensure that when an object (let's call it ObjectA) of their custom resource is created, it references another valid object (let's call it ObjectB). Basically, before ObjectA is created, they want to ensure ObjectB exists.
- Validation Before Deletion: Before deleting an object of their custom resource (let's say ObjectA again), they want to make sure no other objects (e.g., ObjectC, ObjectD, etc.) are referencing it.

Here's a suggested answer:

- Validation During Creation: You might able to achieve it with admission webhook. See: https://book.kubebuilder.io/reference/admission-webhook
- Validation Before Deletion: Then, it seems that you are looking for how to use a finalizer. See an example of code which is scaffold by the deploy image plugin: https://github.com/kubernetes-sigs/kubebuilder/blob/v3.12.0/testdata/project-v4-with-deploy-image/internal/controller/busybox_controller.go#L121-L192

I hope that answer your question so I am closing this one
However, feel free to re-open if you need to do so.
